src/backend/distributed/citus--8.0-1--8.0-2.sql

```diff
@@ -0,0 +1,62 @@
+/* citus--7.5-7--8.0-1 */
+SET search_path = 'pg_catalog';
+
+CREATE OR REPLACE FUNCTION pg_catalog.relation_is_a_known_shard(regclass)
+  RETURNS bool
+LANGUAGE C STRICT
+AS 'MODULE_PATHNAME', $$relation_is_a_known_shard$$;
+COMMENT ON FUNCTION relation_is_a_known_shard(regclass)
+    IS 'returns true if the given relation is a known shard';
+
+CREATE OR REPLACE FUNCTION pg_catalog.citus_table_is_visible(oid)
+  RETURNS bool
+LANGUAGE C STRICT
+STABLE
+PARALLEL SAFE
+AS 'MODULE_PATHNAME', $$citus_table_is_visible$$;
+COMMENT ON FUNCTION citus_table_is_visible(oid)
+	IS 'wrapper on pg_table_is_visible, filtering out tables (and indexes) that are known to be shards';
+
+-- this is the exact same query with what \d
+-- command produces, except pg_table_is_visible
+-- is replaced with pg_catalog.relation_is_a_known_shard(c.oid)
+CREATE VIEW citus.citus_shards_on_worker AS 
+	SELECT n.nspname as "Schema",
+	  c.relname as "Name",
+	  CASE c.relkind WHEN 'r' THEN 'table' WHEN 'v' THEN 'view' WHEN 'm' THEN 'materialized view' WHEN 'i' THEN 'index' WHEN 'S' THEN 'sequence' WHEN 's' THEN 'special' WHEN 'f' THEN 'foreign table' WHEN 'p' THEN 'table' END as "Type",
```

Comment:
Is there a way to encapsulate this in a function ? Same thing is also used in the next view definition

Reply:
What would be the benefit of that?

Reply:
Note that the queries have some minor differences if you're asking to use that function in both views.